SocialCareCaseViewerApi.Tests/V1/Data/MongoDB/UpdateMosaicIdFormDataCollection.cs

```diff
@@ -0,0 +1,156 @@
+
+            var filterByNewId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", masterPersonId);
+            var recordsWithNewId = collection.Find(filterByNewId).ToList();
+            recordsWithNewId.Count.Should().Be(3);
+
+            var filterByOldId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", mosaicId);
+            var recordsWithOldId = collection.Find(filterByOldId).ToList();
+            recordsWithOldId.Count.Should().Be(0);
+        }
+
+        [Test]
+        public void DoesNotUpdateOtherThanMatchingRecords()
+        {
+            string mosaicId = "334445566";
+            string mosaicIdTwo = "556677";
+            string masterPersonId = "77889900";
+
+            var casenoteOne = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+            var casenoteTwo = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+            var casenoteThree = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicIdTwo).Create();
+
+            var collection = _mongoDatabase?.GetCollection<MongoDBTestObject>(MainFormCollection);
+
+            collection?.InsertOne(casenoteOne);
+            collection?.InsertOne(casenoteTwo);
+            collection?.InsertOne(casenoteThree);
+
+            var result = _mongoDatabase?.RunCommand(_dbCommandUnderTest);
+
+            var filterByNewId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", masterPersonId);
+            var recordsWithNewId = collection.Find(filterByNewId).ToList();
+            recordsWithNewId.Count.Should().Be(2);
+
+            var filterByExcludedId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", mosaicIdTwo);
+            var recordsWitExcludedId = collection.Find(filterByExcludedId).ToList();
+            recordsWitExcludedId.Count.Should().Be(1);
+        }
+
+        [Test]
+        public void DoesNotUpdateAuditTypeRecordsThatHaveAPIInTheFormNameOverallField()
+        {
+            string mosaicId = "334445566";
+            string masterPersonId = "77889900";
+
+            var casenoteOne = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).With(x => x.FormNameOverall, "API_Event").Create();
+            var casenoteTwo = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+            var casenoteThree = _fixture.Build<MongoDBTestObject>().With(x => x.MosaicId, mosaicId).Create();
+
+            var collection = _mongoDatabase?.GetCollection<MongoDBTestObject>(MainFormCollection);
+
+            collection?.InsertOne(casenoteOne);
+            collection?.InsertOne(casenoteTwo);
+            collection?.InsertOne(casenoteThree);
+
+            var result = _mongoDatabase?.RunCommand(_dbCommandUnderTest);
+
+            var filterByNewId = Builders<MongoDBTestObject>.Filter.Eq("MosaicId", masterPersonId);
+            var recordsWithNewId = collection.Find(filterByNewId).ToList();
+            recordsWithNewId.Count.Should().Be(2);
```

Comment:
What do you think of changing the expectation to find the case note with the `FormNameOverall` is `API_Event` and checking that the `MosaicId` is still `mosaicId`? I think that's more precise than checking the count.

Reply:
@chubberlisk That's more robust, I've pushed an update.